### PR TITLE
add hyperledger fabric admin functions

### PR DESCRIPTION
**Pull request by yxuco**
Added utility functions for managing channels and chaincodes in hyperledger fabric; integrated "dovetail deploy" for installing to all peers of a specified org. will continue to work on (1) add start command to instantiate chaincode, and (2) provide a REST API service on top of the fabadmin functions.

### CODE REVIEW

File: commands/contract/deploy.go [hunk 1]
```diff
@@ -13,32 +13,56 @@ import (
...
+	ContractCmd.AddCommand(instantiateCmd)
+
 	deployCmd.Flags().StringVar(&id, "id", "", "Id of the Smart Contract")
-	deployCmd.Flags().StringVar(&path, "path", "", "Path of the Smart Contract")
-	deployCmd.Flags().StringVarP(&version, "version", "v", "", "Version of the Smart Contract")
-	deployCmd.Flags().StringVar(&userName, "user", "Admin", "User Name")
-	deployCmd.Flags().StringVar(&orgName, "org", "org1", "Organization Name")
+	deployCmd.Flags().StringVar(&path, "path", "", "Source folder of the generated Smart Contract, e.g., /path/to/hlf/src/myapp")
+	deployCmd.Flags().StringVar(&userName, "user", "Admin", "Admin user Name of the org")
+	deployCmd.Flags().StringVar(&orgName, "org", "", "Organization Name")
```
Comment (torresashjian) [inline commands/contract/deploy.go]: OrgName, config and override seem very HL Fabric specific. We  have a few options to make this blockchain agnostic. Should we have our own manifest that can be blockchain specific so the commands are generic?

File: commands/contract/deploy.go [hunk 2]
```diff
@@ -51,39 +75,69 @@ var deployCmd = &cobra.Command{
...
+				fmt.Println(err)
+				os.Exit(1)
+			}
+		default:
+			fmt.Println("Unsupported blockchain for deploy:", blockchain)
+		}
+	},
+}
+
+var instantiateCmd = &cobra.Command{
```
Comment (torresashjian) [inline commands/contract/deploy.go]: Also here we should think if this "instantiate" concept will be available for all blockchains supported and how does that affect the command structure

File: commands/contract/deploy.go (hunk 1 above)
Comment (yxuco) [reply commands/contract/deploy.go]: it is a problem to make admin commands "blockchain agnostic" while the underneath technologies are completely different.  The "deploy" is assumed the same as "install" for hyperledger fabric, which already makes fabric users a little confused.  Besides, the "generate" command also has a couple of parameters that are corda-specific.  We probably should branch out from the very beginning, so each blockchain has its own commands that makes sense to the underneath technology. e.g., dovetail fabric generate, dovetail fabric install, and dovetail fabric instantiate, and dovetail corda generate, dovetail corda deploy, etc.

File: commands/contract/deploy.go (hunk 2 above)
Comment (yxuco) [reply commands/contract/deploy.go]: ditto

**Pull request merged**